src/core/imap.ts

```diff
@@ -165,28 +143,30 @@ export class IMAP {
 
                     await this.openBox(mailBox);
 
-                    const messages = await this.search(criteria);
-                    const newMessages : number[] = messages.filter((m) => ! this.uids.includes(m));
-
+                    const newMessages = await this.search(criteria);
                     console.log(newMessages, 'new Messages');
 
                     if (newMessages.length && !this.pending) {
                         this.pending = true;
-                        const data = await this.fetchAll(newMessages, {bodies: ''});
 
+                        setupTempFolder = new SetupTempFolder();
```

Comment:
A minor thing but there is no reason to create a new instance on every iteration.
In fact, you don't need to create any instances of this class at all - all method can be made static

Reply:
OK, it will be changed in the next PR.